### PR TITLE
Update deploy.md

### PR DESCRIPTION
**Pull request by snalvi**
Updating documentation to be clearer for `deploymentEnvironment`.

### CODE REVIEW

File: content/en/docs/operator-reference/deploy.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ deploymentEnvironment:
   - `address`: The address of a running [Vault](https://www.vaultproject.io/) datastore. This is only required when Spinnaker is being deployed in non-Kubernetes clustered configuration.
 - `location`: This is the location spinnaker will be deployed to. When deploying to Kubernetes, use this flag to specify the namespace to deploy to (defaults to `spinnaker`)
 - `customSizing`: Configure, validate, and view the component sizings for the Spinnaker services.
-  - `spin-clouddriver`:
+  - `clouddriver`:
```
Comment (german-muzquiz) [inline content/en/docs/operator-reference/deploy.md:98]: `clouddriver` doesn't work, it needs to be `spin-clouddriver`

**Review by german-muzquiz — CHANGES_REQUESTED** [review]

File: content/en/docs/operator-reference/deploy.md (hunk 1 above)
Comment (snalvi) [reply content/en/docs/operator-reference/deploy.md:98]: @german-muzquiz really? My manifest is: 
```
    deploymentEnvironment:
      customSizing:
        clouddriver:
          requests:
            memory: ...
            cpu: ...
          limits:
            memory: ...
            cpu: ...
        deck:
          requests:
            memory: ...
            cpu: ...
          limits:
            memory: ..
            cpu: ...
```
and the values are respected

Comment (snalvi) [reply content/en/docs/operator-reference/deploy.md:98]: When I had the keys as `spin-clouddriver`, `spin-deck` the values did not show up in the deployment manifest.

Comment (german-muzquiz) [reply content/en/docs/operator-reference/deploy.md:98]: I double checked applying this patch: https://github.com/armory/spinnaker-kustomize-patches/blob/master/deploymentEnvironment/patch-sizing.yml after increasing replicas to 2 and I saw 2 pods being created, then I deleted the `spin-` prefix and spinnaker was only left with 1 pod. The above was tested deploying with armory operator `armory/armory-operator:1.1.0` and armory spinnaker `2.21.0`.

How did you deploy spinnaker? I wonder if something changed between versions. According to this Halyard code in master https://github.com/spinnaker/halyard/blob/master/halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Service.java#L184 , replicas are read from the "serviceName" field, which ultimately resolves to `spin-<canonicalName>` according to this: https://github.com/spinnaker/halyard/blob/master/halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/SpinnakerService.java#L196 .

I noticed that other sections like `tolerations` work with or without prefix.

Comment (snalvi) [reply content/en/docs/operator-reference/deploy.md:98]: I just came across this in the Spinnaker documentation:
https://spinnaker.io/reference/halyard/component-sizing/
```
deploymentEnvironment:
  customSizing:
    # This applies sizings to only the clouddriver container and not to any sidecar 
    # containers running with clouddriver.
    clouddriver:
    ...
    # This applies sizings to the clouddriver container as well as any sidecar 
    # containers running with clouddriver.
    spin-clouddriver:
```
So I suppose it's also dependent on how you've setup your deployment, and what section it is. But `spin-clouddriver` and `clouddriver` are both accepting of resource limits with Spinnaker v1.19.9 and armory operator 1.0.2. This ticket can be closed off.

**Pull request closed without merging**